Unsafe is broken

**Describe the bug**

[This](https://github.com/segmentio/kafka-go/pull/1119) development broke unsafe usage,

```
../../go/pkg/mod/github.com/segmentio/kafka-go@v0.4.41/protocol/decode.go:90:4: v.setFloat64 undefined (type value has no field or method setFloat64)
../../go/pkg/mod/github.com/segmentio/kafka-go@v0.4.41/protocol/encode.go:134:19: v.float64 undefined (type value has no field or method float64)
```

We also need to add these functions to reflect_unsafe.go

--- a/protocol/reflect_unsafe.go
+++ b/protocol/reflect_unsafe.go
@@ -63,6 +63,8 @@ func (v value) int32() int32 { return *(*int32)(v.ptr) }
 
 func (v value) int64() int64 { return *(*int64)(v.ptr) }
 
+func (v value) float64() float64 { return *(*float64)(v.ptr) }
+
 func (v value) string() string { return *(*string)(v.ptr) }
 
 func (v value) bytes() []byte { return *(*[]byte)(v.ptr) }
@@ -91,6 +93,8 @@ func (v value) setInt16(i int16) { *(*int16)(v.ptr) = i }
 func (v value) setInt32(i int32) { *(*int32)(v.ptr) = i }
 
 func (v value) setInt64(i int64) { *(*int64)(v.ptr) = i }
+
+func (v value) setFloat64(f float64) { *(*float64)(v.ptr) = i }
 
 func (v value) setString(s string) { *(*string)(v.ptr) = s }
 

--- a/protocol/reflect_unsafe.go
+++ b/protocol/reflect_unsafe.go
@@ -94,7 +94,7 @@ func (v value) setInt32(i int32) { *(*int32)(v.ptr) = i }
 
 func (v value) setInt64(i int64) { *(*int64)(v.ptr) = i }
 
-func (v value) setFloat64(f float64) { *(*float64)(v.ptr) = i }
+func (v value) setFloat64(f float64) { *(*float64)(v.ptr) = f }
 
 func (v value) setString(s string) { *(*string)(v.ptr) = s }
 

